### PR TITLE
 Added filter options to Consumer/Entitlement resource ent lookups

### PR DESCRIPTION
**Pull request by mstead**
Affected API calls:
  GET /consumers/:uuid/entitlements
  GET /entitlements

Caller can now specify an 'attribute' query parameter to filter
results by pool/product attributes.

Caller can now specify a 'matches' query parameter to filter
entitlements by an entitlement pool's:
- product name/id
- contract number
- order number
- provided product name/id

### CODE REVIEW

File: server/client/ruby/candlepin_api.rb [hunk 1]
```diff
@@ -743,10 +743,32 @@ def list_entitlements(params={})
...
 
+  # NOTE: Purely for testing via the entitlement resource.
+  #       Very similar to list_entitlements above (consumer resource)
+  def list_ents_via_entitlements_resource(params={})
+    path = "/entitlements?"
+    path << "consumer=#{params[:consumer_uuid]}&" if params[:consumer_uuid]
+
+    attr_filters = params[:attr_filters] || []
+    attr_filters.each do | attr_name, attr_value |
+      path << "attribute=#{attr_name}:#{attr_value}&"
```
Comment (wottop) [inline server/client/ruby/candlepin_api.rb]: Is a colon a sufficient delimiter? Can an attribute value also have a colon?
Double colon maybe?

**Comment by wottop** [comment]
I assume that the attribute value is match only. No wildcards?

File: server/client/ruby/candlepin_api.rb (hunk 1 above)
Comment (wottop) [inline server/client/ruby/candlepin_api.rb]: I see this 
String[] parts = this.paramValue.split(":", 2); // Maximum of two parts
Guess it shouls work even if the value has a colon. Carry on.

**Comment by wottop** [comment]
Just the one question. Looks good.

**Comment by mstead** [comment]
This filtering works the same as the old Pool attribute filtering in the OwnerResource.

It supports a bunch of different formats. Wildcards are supported. It also handles '![value]' --  attribute=support_level:!Standard -- which means != or doesn't exist.

**Comment by wottop** [comment]
ok, good. 

ack

**Pull request merged**